internal/clusterfeature/clusterfeatureadapter/gorm_feature_repository.go

```diff
@@ -144,3 +155,45 @@ func (r *GormFeatureRepository) modelToFeature(cfm *clusterFeatureModel) (*clust
 
 	return &f, nil
 }
+
+// DeleteFeature permanently deletes the feature record
+func (r *GormFeatureRepository) DeleteFeature(ctx context.Context, clusterId uint, featureName string) error {
+
+	fm := clusterFeatureModel{
+		ClusterID: clusterId,
+		Name:      featureName,
+	}
+
+	if err := r.db.Delete(&fm).Error; err != nil {
```

Comment:
Does this delete the row matching `fm` or all rows? I think we had problems with this elsewhere when not setting the primary key in the query struct.

Reply:
thx, fixed